src/sinks/hdfs/config.rs

```diff
@@ -0,0 +1,192 @@
+    pub name_node: String,
+
+    #[serde(flatten)]
+    pub encoding: EncodingConfigWithFraming,
+
+    #[configurable(derived)]
+    #[serde(default = "Compression::gzip_default")]
+    pub compression: Compression,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    pub batch: BatchConfig<BulkSizeBasedDefaultBatchSettings>,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    pub acknowledgements: AcknowledgementsConfig,
+}
+
+impl GenerateConfig for HdfsConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(Self {
+            root: "/".to_string(),
+            prefix: "%F/".to_string(),
+            name_node: "default".to_string(),
+
+            encoding: (
+                Some(NewlineDelimitedEncoderConfig::new()),
+                JsonSerializerConfig::default(),
+            )
+                .into(),
+            compression: Compression::gzip_default(),
+            batch: BatchConfig::default(),
+
+            acknowledgements: Default::default(),
+        })
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+impl SinkConfig for HdfsConfig {
+    async fn build(&self, _cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let op = self.build_operator()?;
+
+        let check_op = op.clone();
+        let healthcheck = Box::pin(async move { Ok(check_op.check().await?) });
+
+        let sink = self.build_processor(op)?;
+        Ok((sink, healthcheck))
+    }
+
+    fn input(&self) -> Input {
+        Input::new(self.encoding.config().1.input_type() & DataType::Log)
+    }
+
+    fn acknowledgements(&self) -> &AcknowledgementsConfig {
+        &self.acknowledgements
+    }
+}
+
+impl HdfsConfig {
+    pub fn build_operator(&self) -> crate::Result<Operator> {
+        // Build OpenDal Operator
+        let mut builder = Hdfs::default();
+        // Prefix logic will be handled by key_partitioner.
+        builder.root(&self.root);
+        builder.name_node(&self.name_node);
+
+        let op = Operator::create(builder)?
+            .layer(LoggingLayer::default())
+            .finish();
+        Ok(op)
+    }
+
+    pub fn build_processor(&self, op: Operator) -> crate::Result<VectorSink> {
+        // Configure our partitioning/batching.
+        let batcher_settings = self.batch.into_batcher_settings()?;
+
+        let transformer = self.encoding.transformer();
+        let (framer, serializer) = self.encoding.build(SinkType::MessageBased)?;
+        let encoder = Encoder::<Framer>::new(framer, serializer);
+
+        let request_builder = OpenDalRequestBuilder {
+            encoder: (transformer, encoder),
+            compression: self.compression,
+        };
+
+        // TODO: we can add tower middleware here.
+        let svc = ServiceBuilder::new().service(OpenDalService::new(op));
```

Comment:
```suggestion
        // TODO: we can add tower middleware here.
        let svc = ServiceBuilder::new().service(OpenDalService::new(op));
```

**thought**: 🚀 if it's that easy that's great! If all of the OpenDAL based sinks would use the same set of middlewares we'd probably want to end up moving this into the `common` file.